main-command/src/main/scala/sbt/internal/client/NetworkClient.scala

```diff
@@ -375,15 +377,20 @@ class NetworkClient(
                 s"either upgrade $sbtScript to its latest version or make sure it is accessible from $$PATH, and run 'sbt bspConfig'"
               )
             }
-            val java = Option(Properties.javaHome)
+            val java = javaHomeOpt
               .map { javaHome =>
                 s"$javaHome/bin/java"
               }
               .getOrElse("java")
             List(java) ++ arguments.sbtArguments ++
               List("-jar", lj, DashDashDetachStdio, DashDashServer)
           case _ =>
-            List(arguments.sbtScript) ++ arguments.sbtArguments ++
+            // From sbt/sbt.sh/sbt.bat:
+            //    --java-home <path>         alternate JAVA_HOME
+            val javaHomeArgs = javaHomeOpt.toSeq.flatMap(javaHome => Seq("--java-home", javaHome))
```

Comment:
Note that `java.home` system properties, even though it is named as such, isn't the same as `JAVA_HOME`.
See https://docs.oracle.com/javase/tutorial/essential/environment/sysprop.html
> `"java.home"`: Installation directory for Java Runtime Environment (JRE)

This means that `java` would be on the path, but if I recall correctly, `javac` will not be on the path, which could break Java compilation.

Reply:
We are passing in the script location via system properties, like

```
"-Dsbt.script=/Users/xxx/.sdkman/candidates/sbt/current/bin/sbt"
```

so we could pass in `JAVA_HOME` as `-Dsbt.java.home` and use that when present.